### PR TITLE
Use rspec-puppet-facts

### PR DESCRIPTION
**Pull request by mcanevet**
Work in Progress. Could be merged as it though.

### CODE REVIEW

File: spec/classes/foreman_config_passenger_spec.rb [hunk 1]
```diff
@@ -1,168 +1,132 @@
...
-        })
-      end
-
-      it 'should not include start timeout fragment on Squeeze' do
-        should contain_apache__vhost('foreman-ssl').without({
-          :custom_fragment => %r{^PassengerStartTimeout},
-        })
-      end
-
-      it 'should not include the Ruby interpreter' do
```
Comment (domcleal) [inline spec/classes/foreman_config_passenger_spec.rb]: Any way we can keep this example for wheezy and trusty?

**Comment by mcanevet** [comment]
@domcleal I removed `foreman::config::passenger`'s specs refactoring from the PR as it is not ready yet.

**Comment by domcleal** [comment]
Ah okay, that's fine for now, we can have another go later.  Could you give it a rebase now I've done the modulesync?  I think we're ready to go.

**Comment by ekohl** [comment]
:+1: when the tests pass

**Comment by domcleal** [comment]
They're taking a very long time!

**Comment by ekohl** [comment]
That's caused by running the tests on many operatingsystems since there are 11 in metadata.json. I suspect that ```include foreman``` as precondition in many tests makes it a lot slower than needed.

**Comment by domcleal** [comment]
Thanks @mcanevet, great work!

**Comment by mcanevet** [comment]
\o/

**Pull request merged**